gensim/models/base_any2vec.py

```diff
@@ -330,12 +334,14 @@ def __init__(self, sentences=None, workers=3, vector_size=100, epochs=5, callbac
                 self.neg_labels[0] = 1.
 
         if sentences is not None:
-            if isinstance(sentences, GeneratorType):
+            if multistream and not isinstance(sentences, (tuple, list)):
+                raise TypeError("If multistream=True, you must pass tuple or list as the sentences argument.")
+            if not multistream and isinstance(sentences, GeneratorType):
```

Comment:
This check (about generators) should be applied for each multistream element too (i.e. to each stream)